### PR TITLE
[#55] Implement default image stub: init

### PR DESCRIPTION
**Pull request by GetFire**

### CODE REVIEW

File: src/main/java/net/greatstart/services/IdenticonGeneratorService.java [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+                }
+                raster.setPixel(x, y, pixelColor);
+            }
+        }
+
+        BufferedImage finalImage = new BufferedImage(imageWidth, imageHeight, BufferedImage.TYPE_INT_ARGB);
+
+        //Scale image to the size you want
+        AffineTransform at = new AffineTransform();
+        at.scale(imageWidth / width, imageHeight / height);
```
Comment (Sheptytskyid) [inline src/main/java/net/greatstart/services/IdenticonGeneratorService.java]: MINOR Cast one of the operands of this division operation to a "double". [rule](https://sonarqube.com/coding_rules#rule_key=squid%3AS2184)
MINOR Cast one of the operands of this division operation to a "double". [rule](https://sonarqube.com/coding_rules#rule_key=squid%3AS2184)

File: src/main/java/net/greatstart/services/IdenticonGeneratorService.java [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    public static byte[] generateByteImage(String text, int imageWidth, int imageHeight) {
+        BufferedImage image = generateImage(text, imageWidth, imageHeight);
+        byte[] photoBytes = null;
+        try {
+            ByteArrayOutputStream baos = new ByteArrayOutputStream();
+            ImageIO.write(image, "jpg", baos);
+            baos.flush();
+            photoBytes = baos.toByteArray();
+        } catch (IOException e) {
+            e.printStackTrace();
```
Comment (Sheptytskyid) [inline src/main/java/net/greatstart/services/IdenticonGeneratorService.java]: MINOR Use a logger to log this exception. [rule](https://sonarqube.com/coding_rules#rule_key=squid%3AS1148)

**Comment by Sheptytskyid** [comment]
SonarQube analysis reported 3 issues
* ![MINOR][MINOR] 3 minor

Watch the comments in this conversation to review them.

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: src/main/java/net/greatstart/services/IdenticonGeneratorService.java [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+                raster.setPixel(x, y, pixelColor);
+            }
+        }
+
+        BufferedImage finalImage = new BufferedImage(imageWidth, imageHeight, BufferedImage.TYPE_INT_ARGB);
+
+        //Scale image to the size you want
+        AffineTransform at = new AffineTransform();
+        at.scale(imageWidth / width, imageHeight / height);
+        AffineTransformOp op = new AffineTransformOp(at, AffineTransformOp.TYPE_NEAREST_NEIGHBOR);
```
Comment (Sheptytskyid) [inline src/main/java/net/greatstart/services/IdenticonGeneratorService.java]: MINOR Cast one of the operands of this division operation to a "double". [rule](https://sonarqube.com/coding_rules#rule_key=squid%3AS2184)
MINOR Cast one of the operands of this division operation to a "double". [rule](https://sonarqube.com/coding_rules#rule_key=squid%3AS2184)

File: src/main/java/net/greatstart/services/IdenticonGeneratorService.java [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+        BufferedImage image = generateImage(text, imageWidth, imageHeight);
+        byte[] photoBytes = null;
+        try {
+            ByteArrayOutputStream baos = new ByteArrayOutputStream();
+            ImageIO.write(image, "jpg", baos);
+            baos.flush();
+            photoBytes = baos.toByteArray();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
```
Comment (Sheptytskyid) [inline src/main/java/net/greatstart/services/IdenticonGeneratorService.java]: MINOR Use a logger to log this exception. [rule](https://sonarqube.com/coding_rules#rule_key=squid%3AS1148)

File: src/main/java/net/greatstart/services/IdenticonGeneratorService.java [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+                }
+                raster.setPixel(x, y, pixelColor);
+            }
+        }
+
+        BufferedImage finalImage = new BufferedImage(imageWidth, imageHeight, BufferedImage.TYPE_INT_ARGB);
+
+        //Scale image to the size you want
+        AffineTransform at = new AffineTransform();
+        at.scale(imageWidth / width, imageHeight / height);
```
Comment (Sheptytskyid) [inline src/main/java/net/greatstart/services/IdenticonGeneratorService.java]: MINOR Cast one of the operands of this division operation to a "double". [rule](https://sonarqube.com/coding_rules#rule_key=squid%3AS2184)
MINOR Cast one of the operands of this division operation to a "double". [rule](https://sonarqube.com/coding_rules#rule_key=squid%3AS2184)

File: src/main/java/net/greatstart/services/IdenticonGeneratorService.java [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+    public static byte[] generateByteImage(String text, int imageWidth, int imageHeight) {
+        BufferedImage image = generateImage(text, imageWidth, imageHeight);
+        byte[] photoBytes = null;
+        try {
+            ByteArrayOutputStream baos = new ByteArrayOutputStream();
+            ImageIO.write(image, "jpg", baos);
+            baos.flush();
+            photoBytes = baos.toByteArray();
+        } catch (IOException e) {
+            e.printStackTrace();
```
Comment (Sheptytskyid) [inline src/main/java/net/greatstart/services/IdenticonGeneratorService.java]: MINOR Use a logger to log this exception. [rule](https://sonarqube.com/coding_rules#rule_key=squid%3AS1148)

**Comment by Sheptytskyid** [comment]
SonarQube analysis reported 2 issues
* ![MINOR][MINOR] 2 minor

Watch the comments in this conversation to review them.

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: src/main/java/net/greatstart/dto/DtoUserProfile.java [hunk 7]
```diff
@@ -16,4 +16,5 @@
     private String email;
     private String lastName;
     private byte[] photo;
+    private String initial;
```
Comment (Sheptytskyid) [inline src/main/java/net/greatstart/dto/DtoUserProfile.java:19]: MAJOR Remove this unused "initial" private field. [rule](https://sonarqube.com/coding_rules#rule_key=squid%3AS1068)

**Comment by Sheptytskyid** [comment]
SonarQube analysis reported 1 issue
* ![MAJOR][MAJOR] 1 major

Watch the comments in this conversation to review them.

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Review by zozich — APPROVED** [review]
Ok, but I would rather implement the logic on front-end only

**Pull request merged**